b2sdk/raw_simulator.py

```diff
@@ -831,6 +877,30 @@ def hide_file(self, api_url, account_auth_token, bucket_id, file_name):
         self.file_id_to_bucket_id[response['fileId']] = bucket_id
         return response
 
+    def copy_file(
+        self,
+        api_url,
+        account_auth_token,
+        file_id,
+        new_file_name,
+        bytes_range=None,
+        metadata_directive=None,
+        content_type=None,
+        file_info=None,
+    ):
+        bucket_id = self.file_id_to_bucket_id[file_id]
+        bucket = self._get_bucket_by_id(bucket_id)
+        self._assert_account_auth(api_url, account_auth_token, bucket.account_id, 'writeFiles')
```

Comment:
No need to readFiles?

Reply:
I don't know how these permissions work. They should write in API doc which permissions are required for each of the API.

Reply:
please ask on slack, tag Brian G